sdk/core/azure-core/test/ut/transport_adapter_curl.cpp

```diff
@@ -36,89 +37,131 @@ namespace Azure { namespace Core { namespace Test {
     }
   } // namespace
 
-/***********************  Unique Tests for Libcurl   ********************************/
-//  Disabling test with INCLUDE_DISABLED_TESTS. The test name cannot be changed because it depends
-//  on a friend class definition. Hence, it can't use the gtest DISABLE_.
-#if defined(INCLUDE_DISABLED_TESTS)
-    TEST_P(TransportAdapter, connectionPoolTest)
+  /***********************  Unique Tests for Libcurl   ********************************/
+  TEST_P(TransportAdapter, connectionPoolTest)
+  {
+    Azure::Core::Http::CurlConnectionPool::ClearIndex();
+
+    // Make sure there are nothing in the pool
+    EXPECT_EQ(Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.size(), 0);
+
+    // Use the same request for all connections.
+    Azure::Core::Http::Request req(
+        Azure::Core::Http::HttpMethod::Get, Azure::Core::Http::Url("http://httpbin.org/get"));
+
     {
-      Azure::Core::Http::Url host("http://httpbin.org/get");
-      Azure::Core::Http::CurlConnectionPool::ClearIndex();
-
-      auto threadRoutine = [&]() {
-        using namespace std::chrono_literals;
-        auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, host);
-        auto response = m_pipeline->Send(Azure::Core::GetApplicationContext(), request);
-        checkResponseCode(response->GetStatusCode());
-        auto expectedResponseBodySize = std::stoull(response->GetHeaders().at("content-length"));
-        CheckBodyFromBuffer(*response, expectedResponseBodySize);
-        std::this_thread::sleep_for(1s);
-      };
```

Comment:
The previous test is not coming back.  It can't be deterministic because it depends on the actual network.
I need to control exactly when a connection is returned to the pool, and by sending a request I can't. It can take nothing to send the request and release the connection, or it could take much more.